kratos/python_scripts/json_output_process.py

```diff
@@ -10,177 +10,444 @@ def Factory(settings, Model):
+                    elif variable_type == "Vector":
+                        if (self.resultant_solution == False):
+                            for gp in range(gauss_point_number):
+                                list = self.__kratos_vector_to__python_list(value[gp])
+                                data["ELEMENT_" + str(elem.Id)][variable_name][str(gp)].append(list)
+                        else:
+                            if (count == 0):
+                                for gp in range(gauss_point_number):
+                                    list = self.__kratos_vector_to__python_list(value[gp])
+                                    data["RESULTANT"][variable_name][str(gp)][-1] += list
+
+                        # TODO: Add pending classes
                 count += 1
-              
+
         write_external_json(self.output_file_name, data)
 
     def ExecuteBeforeOutputStep(self):
+        """ This method is executed right before the ouput process computation
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+
         pass
 
     def ExecuteAfterOutputStep(self):
+        """ This method is executed right after the ouput process computation
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+
         pass
 
     def ExecuteFinalize(self):
+        """ This method is executed in order to finalize the current computation
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+
         pass
-      
-    def __kratos_vector_to__python_list(self,value):
+
+    def __kratos_vector_to__python_list(self, value):
+        """ This method is executed in order to finalize the current computation
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        value -- The Kratos vector to transform
+        """
+
         list = []
         for index in range(len(value)):
             list.append(value[index])
         return list
-        
-    def __generate_variable_list_from_input(self,param):
-      '''Parse a list of variables from input.'''
+
+    def __generate_variable_list_from_input(self, param):
+      """ Parse a list of variables from input.
+
+      Keyword arguments:
+      self -- It signifies an instance of a class.
+      value -- The Kratos vector to transform
+      """
+
       # At least verify that the input is a string
       if not param.IsArray():
           raise Exception("{0} Error: Variable list is unreadable".format(self.__class__.__name__))
 
       # Retrieve variable name from input (a string) and request the corresponding C++ object to the kernel
-      return [ KratosMultiphysics.KratosGlobals.GetVariable( param[i].GetString() ) for i in range( 0,param.size() ) ]
+      return [KratosMultiphysics.KratosGlobals.GetVariable(param[i].GetString()) for i in range( 0, param.size())]
 
+    def __check_variable_type(self, variable):
```

Comment:
I saw this being used several times already. Maybe you could add it to KratosGlobals? => `GetVariableType`

Just a recommendation, it would be cool to have it there, not a blocker though

Reply:
Huuum, I can...